File: backend/src/main/java/com/harpiaCrud/backend/model/Order.java
package com.harpiaCrud.backend.model;

import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

import jakarta.persistence.Entity;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "client_order")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @ManyToOne
    private Client client;

    private double totalPrice;
}

File: frontend/src/app/interfaces/order/order.ts
import { Client } from "../client/client";
import { Product } from "../product/product";

interface Product_Order {
    qt: number,
    product: Product,
    order: Order
}

export interface Order {
    id?: string,
    client: Client,
    products: Product_Order[],
    totalPrice: number
}